packages/notifi-dapp-example/src/context/GlobalStateContext.tsx

```diff
@@ -16,23 +16,51 @@ import {
 export type GlobalStateContextType = {
   isGlobalLoading: boolean;
   setIsGlobalLoading: Dispatch<SetStateAction<boolean>>;
+  /* deprecated: use globalInfoModal instead */
   globalError: null | string;
+  /* deprecated: use setGlobalInfoModal instead */
   setGlobalError: Dispatch<SetStateAction<string | null>>;
+  popGlobalInfoModal: (globalInfoModal: GlobalInfoModal) => void;
 };
 
 const GlobalStateContext = createContext<GlobalStateContextType>({
   isGlobalLoading: false,
   setIsGlobalLoading: () => undefined,
+  /* deprecated: use globalInfoModal instead */
   globalError: null,
+  /* deprecated: use setGlobalInfoModal instead */
   setGlobalError: () => undefined,
+  popGlobalInfoModal: () => undefined,
 });
 
+export type GlobalInfoModal = {
+  message: string;
+  iconOrEmoji: SpriteIcon | Emoji;
+  timeout: number /* in milliseconds */;
+};
+
+type SpriteIcon = { type: 'icon'; id: SpriteIconId };
+type Emoji = { type: 'emoji'; content: string };
+
 export const GlobalStateContextProvider: FC<PropsWithChildren> = ({
   children,
 }) => {
   const [isGlobalLoading, setIsGlobalLoading] = useState(false);
+
   const [globalError, setGlobalError] = useState<string | null>(null);
+  const [globalInfoModal, setGlobalInfoModal] =
+    useState<GlobalInfoModal | null>(null);
 
+  const popGlobalInfoModal = (modalData: Partial<GlobalInfoModal>) => {
```

Comment:
just be curious, why we use the popGlobalInfoModal here but not only use setGlobalInfoModal for showing the global info

Reply:
the name start with `set` will be normally considered as native react hook useState with type of `Dispatch`. To avoid that, we name it with `pop` to represent the action that we want to populate the modal CC: @marukohao